pkg/util/tracing/collector/collector.go

```diff
@@ -53,59 +50,40 @@ func New(
 	}
 }
 
-// GetSpanRecordingsFromCluster returns the inflight span recordings from all
-// nodes in the cluster.
+// GetNodesForTraceCollection initializes the TraceCollector by identifying all
+// non-decommissioned nodes in the cluster.
+func (t *TraceCollector) GetNodesForTraceCollection(ctx context.Context) ([]roachpb.NodeID, error) {
```

Comment:
Discussed a bit offline, but it's a bit odd to surface the trace collector's view of node liveness. I think the iterator pattern is a good target to aim for, but this isn't quite there. I suggest trying to somehow work in an API like:

```go
for iter := tcollector.Start(); iter.Valid(); iter.Next() { // .Valid() capturs errors + no more spans
   _ = iter.Value() // tracingpb.RecordedSpan
}
if err := iter.Error(); err != nil {
  return err
}
```

The type backing iter could internally maintain a list of nodes it needs to go through + the index it's currently at, cache the set of recorded spans it collected from the last node + the index it's currently at, and the error it's encountered, if any.

Reply:
Makes sense, I guess we give up on the ability to parallelize fetching these spans from all nodes, but switching to this streaming nature makes sure we don't buffer more than a nodes span recordings at once. Since this is primarily going to be an observability tool, I think that's fine.